components/style/properties_and_values.rs

```diff
@@ -0,0 +1,1167 @@
+                        $input.expect_ident_cloned().and_then(|y| {
+                            if idents_eq(&**x, &*y) {
+                                Ok(SpecifiedValueItem::SpecificIdent(x.clone()))
+                            } else {
+                                Err(BasicParseError::UnexpectedToken(Token::Ident(y)).into())
+                            }
+                        }).map_err(|e| e.into())
+                    },
+                }
+            };
+        }
+
+        fn parse_custom_ident<'i, 't>(
+            _context: &ParserContext,
+            input: &mut Parser<'i, 't>,
+        ) -> Result<Ident, StyleTraitsParseError<'i>> {
+            input.expect_ident_cloned()
+                 .map(|x| Ident((&*x).into()))
+                 .map_err(|e| e.into())
+        }
+
+        parse! {
+            *self, context, input,
+
+            Length => specified::Length::parse,
+            Number => specified::Number::parse,
+            Percentage => specified::Percentage::parse,
+            LengthPercentage => specified::LengthOrPercentage::parse,
+            Color => specified::Color::parse,
+            Image => specified::Image::parse,
+            Url => specified::url::SpecifiedUrl::parse,
+            Integer => specified::Integer::parse,
+            Angle => specified::Angle::parse,
+            Time => specified::Time::parse,
+            TransformList => transform::parse,
+            CustomIdent => parse_custom_ident
+        }
+    }
+}
+
+
+/// A custom property syntax string that is either some basic syntax string
+/// (e.g. some <url> value) or some list term. A list term syntax string allows
+/// a space-separated list of one or more repetitions of the type specified by
+/// the string. Used to build up disjunctions.
+#[derive(Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub struct Term {
+    /// The type of the term (e.g. <integer>).
+    pub typ: Type,
+    /// Whether or not the term is a list, i.e., if the syntax string was
+    /// <integer>+.
+    pub list: bool,
+}
+
+/// A custom property syntax string.
+#[derive(Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum Syntax {
+    /// Syntax to allow any token stream (written '*').
+    Anything,
+    /// Syntax to allow some disjunction of terms (possibly list terms), which
+    /// allows any value matching one of the items in the combination, matched
+    /// in specified order (written 'a | b | ...').
+    Disjunction(Vec<Term>),
+}
+
+impl Syntax {
+    /// Parse a syntax string given to `CSS.registerProperty`.
+    pub fn from_string(input: &str) -> Result<Syntax, ()> {
+        // Syntax strings are DOMStrings, but in Servo we assume they are valid
+        // UTF-8. See
+        // https://doc.servo.org/script/dom/bindings/str/struct.DOMString.html
+        // . This justifies iteration by |char|.
+
+        // Can identifiers in syntax strings contain escapes? No.
```

Comment:
Ok, this means that what I pointed out about `Ident` is not a bug, I think. Still I'd prefer if we re-used the `CustomIdent` type, wdyt?

Reply:
Sounds good! Thanks for noticing that! :D